github-webhook-util/Main.hs

```diff
@@ -0,0 +1,274 @@
+
+getStatuses :: HasGHAuth => GH.Name GH.Owner -> GH.Name GH.Repo -> GH.Commit -> IO (Vector GH.Status)
+getStatuses owner repo commit = do
+  statusCompleted <- checkStatusCompleted owner repo commit
+  possibleStatuses <- runExceptT $ runMonad $ githubRequestRW $ GH.statusesForR owner repo (GH.commitSha commit) GH.FetchAll
+  if not statusCompleted then pure V.empty else
+    case possibleStatuses of
+      (Right statuses) -> pure statuses
+      _                -> pure V.empty
+
+processStatuses :: (GH.Name GH.Commit, AccumulatorValue) -> IO ()
+processStatuses (nameCommit, AccumulatorValue _commit statuses firstCreatedAt) = do
+  let
+      database :: Database
+      database = "ci_statistics"
+      measurement :: Measurement
+      measurement = "ci_statistics"
+      time = firstCreatedAt
+      tags :: [ ( Key, Key ) ]
+      tags = [
+          ("commit", Key $ GH.untagName nameCommit)
+        --, ("pr", Key $ tshow prnum)
+        ]
+      fields :: [ ( Key, LineField ) ]
+      fields = V.toList $ V.map (statusToField time) statuses
+      entry = Line measurement
+             (Map.fromList tags)
+             (Map.fromList fields)
+             time
+  if V.length statuses > 0 then
+    write (writeParams database) entry
+  else
+    pure ()
+
+statusToField :: Maybe UTCTime -> GH.Status -> (Key, LineField)
+statusToField createdAt status = statusField
+  where
+    statusField :: ( Key, LineField )
+    statusField = ( Key context, FieldInt timeInSeconds )
+    context = fromJust $ GH.statusContext status
+    timeInSeconds :: Int64
+    timeInSeconds = round $ diffUTCTime statusUpdatedAt justCreatedAt
+    statusUpdatedAt = GH.statusUpdatedAt status
+    justCreatedAt = fromJust createdAt
+
+manualInfluxImport :: IO ()
+manualInfluxImport = do
+  manager <- newManager tlsManagerSettings
+  auth' <- getAuth
+  let
+      owner :: GH.Name GH.Owner
+      owner = "input-output-hk"
+      repo :: GH.Name GH.Repo
+      repo = "cardano-sl"
+  case auth' of
+    Nothing -> return ()
+    Just auth -> withGHAuth (manager, auth) $ do
+      pullRequests <- getPullRequests owner repo
+      acc <- V.foldM' (foldRequestsToStatuses owner repo) (Accumulator HM.empty) (V.take 2 pullRequests)
+      mapM_ processStatuses $ HM.toList $ ghStatusMap acc
+
+influxImportCommit :: HasGHAuth => GH.Name GH.Owner -> GH.Name GH.Repo -> GH.Name GH.Commit -> IO ()
+influxImportCommit owner repo commitSha = do
+    possibleCommit <- getCommit owner repo commitSha
+    case possibleCommit of
+      Left _       -> putStrLn "An error occurred loading the commit"
+      Right commit -> do
+        acc <- foldCommitsToStatuses owner repo (Accumulator HM.empty) commit
+        putStrLn "Importing commit"
+        mapM_ processStatuses $ HM.toList $ ghStatusMap acc
+
+main :: IO ()
+main = do
+  self <- getProgName
+  let
+      go :: String -> IO ()
+      go "servant" = servantMain
+      go "manual"  = manualInfluxImport
```

Comment:
if you adjust these names, and then make a symlink in the `$out/bin` then you can access both mains